pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -556,7 +582,8 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication {
         removedUnderlyingBalanceRaw = (bufferBalances.getBalanceRaw() * sharesToRemove) / totalShares;
         removedWrappedBalanceRaw = (bufferBalances.getBalanceDerived() * sharesToRemove) / totalShares;
 
-        _supplyCredit(IERC20(_bufferAssets[wrappedToken]), removedUnderlyingBalanceRaw);
+        IERC20 underlyingToken = IERC20(_bufferAssets[wrappedToken]);
```

Comment:
Consider adding a comment here about why we're not calling `asset` on the wrapped token. We've guaranteed the asset on the way in, so we can trust the Vault's token assignment on the way out.